src/rgw/rgw_common.h

```diff
@@ -232,6 +232,9 @@ using ceph::crypto::MD5;
 #define UINT32_MAX (0xffffffffu)
 #endif
 
+// This should not clash with the various user types(RGWUserSourceType)
+#define TYPE_ROLE 4
```

Comment:
why can't this one be in `enum RGWUserSourceType`?

Reply:
A Role isn't a user, so thought of making this as a separate type.

Reply:
i just worry that separating them makes it hard to enforce the 'this should not clash' part. can we rename the enum instead, so that a role type makes sense?

Reply:
RGWIdentityType?